src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -34,14 +35,23 @@ internal class CompletionHandler : IRequestHandler<LSP.CompletionParams, object>
 
             var position = await document.GetPositionFromLinePositionAsync(ProtocolConversions.PositionToLinePosition(request.Position), cancellationToken).ConfigureAwait(false);
 
+            // Filter out unimported types for now as there are two issues with providing them:
+            // 1.  LSP client does not currently provide a way to provide detail text on the completion item to show the namespace.
+            //     https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1076759
+            // 2.  We need to figure out how to provide the text edits along with the completion item or provide them in the resolve request.
+            //     https://devdiv.visualstudio.com/DevDiv/_workitems/edit/985860/
+            var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var completionOptions = documentOptions.WithChangedOption(CompletionServiceOptions.IsExpandedCompletion, false);
```

Comment:
You should use `CompletionOptions.ShowItemsFromUnimportedNamespaces` instead, which is corresponding to the option to enable/disable import completion in VS. `IsExpandedCompletion` is a flag used to indicate that expander is explicitly selected in completion list.

Reply:
I think the first step for LSP to support import completion is to implement expander. Basically, we don't show unimported items by default, but when user select expander, it make a request by setting `IsExpandedCompletion` to true.